src/offline/cable_output.F90

```diff
@@ -1935,239 +1979,260 @@ SUBROUTINE write_output(dels, ktau, met, canopy, casaflux, casapool, casamet, ss
+    LOGICAL, INTENT(IN) :: output_var ! Whether to write the current variable
+    INTEGER, INTENT(IN) :: varID ! variable's netcdf ID
+    CHARACTER(LEN=*), INTENT(IN) :: vname ! name of variable
+    REAL(4), INTENT(IN) :: out_var(:)
+    REAL(4), INTENT(IN) :: acc_val(:)
+    REAL, INTENT(IN) :: vrange(2) ! max and min for variable
+    LOGICAL, INTENT(IN) :: writepatch ! write patch-specific info for this var?
+    TYPE(output_var_settings_type), INTENT(IN) :: out_settings ! output specific settings
+
+    IF ((check%ranges .EQ. ON_TIMESTEP) .OR. (out_settings%writenow .AND. (check%ranges .EQ. ON_WRITE))) THEN
+      CALL check_range(vname, acc_val, vrange, out_timestep, out_settings%met)
+    END IF
+
+    IF (output_var .AND. out_settings%writenow) THEN
+      ! Write value to file:
+      CALL write_ovar(out_timestep, ncid_out, varID, vname, &
+                      out_var, writepatch, out_settings%dimswitch, out_settings%met)
+    END IF
   END SUBROUTINE check_and_write_d1
 
-  SUBROUTINE check_and_write_d2(output_var, varID, vname, out_var, acc_val, vrange, writepatch, met)
-     !! TODO: Remove met
-     LOGICAL, INTENT(IN) :: output_var ! Whether to write the current variable
-     INTEGER, INTENT(IN) :: varID ! variable's netcdf ID
-     CHARACTER(LEN=*), INTENT(IN) :: vname ! name of variable
-     REAL(4), DIMENSION(:,:) :: out_var
-     REAL(4), DIMENSION(:,:) :: acc_val
-     REAL, DIMENSION(2), INTENT(IN) :: vrange ! max and min for variable
-     LOGICAL, INTENT(IN) :: writepatch ! write patch-specific info for this var?
-     TYPE(met_type), INTENT(IN) :: met  ! met data
-
-     IF ((check%ranges .EQ. ON_TIMESTEP) .OR. (writenow .AND. (check%ranges .EQ. ON_WRITE))) THEN
-          CALL check_range(vname, acc_val, vrange, out_timestep, met)
-     END IF
-
-     IF (output_var .AND. writenow) THEN
-          ! Write value to file:
-          CALL write_ovar(out_timestep, ncid_out, varID, vname, &
-               out_var, writepatch, dimswitch, met)
-          ! Reset temporary output variable:
-          out_var = 0.0
-     END IF
+  SUBROUTINE check_and_write_d2(output_var, varID, vname, out_var, acc_val, vrange, writepatch, out_settings)
+    LOGICAL, INTENT(IN) :: output_var ! Whether to write the current variable
+    INTEGER, INTENT(IN) :: varID ! variable's netcdf ID
+    CHARACTER(LEN=*), INTENT(IN) :: vname ! name of variable
+    REAL(4), INTENT(IN) :: out_var(:, :)
+    REAL(4), INTENT(IN) :: acc_val(:, :)
+    REAL, INTENT(IN) :: vrange(2) ! max and min for variable
+    LOGICAL, INTENT(IN) :: writepatch ! write patch-specific info for this var?
+    TYPE(output_var_settings_type), INTENT(IN) :: out_settings  ! output specific settings
+
+    IF ((check%ranges .EQ. ON_TIMESTEP) .OR. (out_settings%writenow .AND. (check%ranges .EQ. ON_WRITE))) THEN
+      CALL check_range(vname, acc_val, vrange, out_timestep, out_settings%met)
```

Comment:
Since we are calling `check_range` from a function that has various forms depending on the array size, we could simplify `check_range()` to be an elemental function and call `range_check()` from here.

But I think it complicates `contant_check_range()` so it's probably not worth it at this stage. It might make more sense once we reorganise the data structures.

Reply:
How would it complicate `constant_check_range`?

We would also change a lot in `write_ovar` as well, which we could do in another issue

Reply:
In constant_check_range, we call check_range for each parameter. If we change to an ELEMENTAL function, then we need 2 lines per parameter in constant_check_range instead of one: 
- call to check_range
- call to range_abort

So I'm not sure it is worth doing it now.

Reply:
I think `check_range` can call `range_abort` if we make all arguments as scalar - `met_type` and `var_range` are the only ones. I also think we can do this in the future.